### PR TITLE
LPD-1377 Added keyboard events so filter can be sorted

### PR DESCRIPTION
**Pull request by fortunatomaldonado**
https://liferay.atlassian.net/browse/LPD-1377

I added keyboard events and utilized `onDragCrossover` and `onDrop` to rearrange the filters. I also included a color change when the filters are being rearranged to better see which filter is moving.

Please let me know if there are any question or comments about this.
Thank you!

[Screencast from 04-19-2024 01:47:21 PM.webm](https://github.com/liferay/liferay-portal/assets/31672551/a82b8fe0-5a55-464d-833d-234391134d29)

### CODE REVIEW

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/css/OrderableTable.scss [hunk 1]
```diff
@@ -27,6 +27,10 @@ html#{$cadmin-selector} .cadmin {
 
 		.orderable-table-row.dragging {
 			opacity: 0.5;
+
+			td.drag-handle-cell {
+				background-color: #528eff !important;
```
Comment (bryceosterhaus) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/css/OrderableTable.scss:32]: Is there a clay variable for this? I noticed this in the clay repo, https://github.com/liferay/clay/blob/5b42133dfbde7d1f2141f558202893394a284de8/packages/clay-tabs/docs/markup-tabs.md?plain=1#L1218

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/OrderableTable.tsx [hunk 2]
```diff
@@ -124,12 +183,22 @@ const Row = ({
...
+				) : null}
+
 				<ClayButtonWithIcon
 					aria-label={Liferay.Util.sub(
 						Liferay.Language.get('drag-x'),
 						item.label || Liferay.Language.get('item')
 					)}
 					displayType={null}
+					onBlur={() => onBlur()}
+					onKeyDown={(event) => onKeyDown(event)}
```
Comment (bryceosterhaus) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/OrderableTable.tsx:201]: You can just do this
```suggestion
					onBlur={onBlur}
					onKeyDown={onKeyDown}
```

**Review by bryceosterhaus — COMMENTED** [review]
Looks good! Just two nits

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/css/OrderableTable.scss (hunk 1 above)
Comment (fortunatomaldonado) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/css/OrderableTable.scss:32]: I looked into this and I'm not sure how to import this clay variable in the file. 
Let me know if you have an idea how to go about doing this.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/OrderableTable.tsx (hunk 2 above)
Comment (fortunatomaldonado) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/OrderableTable.tsx]: Made the changes!

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/css/OrderableTable.scss (hunk 1 above)
Comment (bryceosterhaus) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/css/OrderableTable.scss:32]: It looks like that value isn't in the clay/css source, only the docs, so I think this should be fine.

**Review by bryceosterhaus — APPROVED** [review]
LGTM!

**Comment by fortunatomaldonado** [comment]
ci:forward

**Pull request closed without merging**